client/extensions/woocommerce/state/sites/settings/general/handlers.js

```diff
@@ -0,0 +1,53 @@
+/**
+ * Internal dependencies
+ */
+import { areSettingsGeneralLoaded } from 'woocommerce/state/sites/settings/general/selectors';
+import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
+import { http } from 'state/data-layer/wpcom-http/actions';
+import { setError } from '../../status/wc-api/actions';
+import {
+	WOOCOMMERCE_SETTINGS_GENERAL_REQUEST,
+	WOOCOMMERCE_SETTINGS_GENERAL_REQUEST_SUCCESS,
+} from 'woocommerce/state/action-types';
+
+export const handleSettingsGeneralSuccess = ( { dispatch }, action, next, { data } ) => {
+	const { siteId } = action;
+	dispatch( {
+		type: WOOCOMMERCE_SETTINGS_GENERAL_REQUEST_SUCCESS,
```

Comment:
It *feels* wrong that we have these actions:
* `WOOCOMMERCE_SETTINGS_GENERAL_REQUEST`: When starting the request.
* `WOOCOMMERCE_SETTINGS_GENERAL_REQUEST_SUCCESS`: When completing the request successfully.
* `WOOCOMMERCE_ERROR_SET` (with the original `WOOCOMMERCE_SETTINGS_GENERAL_REQUEST` action as a prop) when there is an error.

That seems inconsistent. Why not have an action called `WOOCOMMERCE_API_SUCCESS` that has the original `WOOCOMMERCE_SETTINGS_GENERAL_REQUEST` action as a prop?